module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php

```diff
@@ -0,0 +1,287 @@
+     * Status code for items with uncertain availability
+     *
+     * @var int
+     */
+    public const STATUS_UNCERTAIN = 2;
+
+    /**
+     * Items availability
+     *
+     * @var int
+     */
+    protected int $availability;
+
+    /**
+     * If unknown message shall be used
+     *
+     * @var bool
+     */
+    protected bool $useUnknownMessage = false;
+
+    /**
+     * Constructor
+     *
+     * @param int|bool $availability Availability
+     * @param string   $status       Status Description
+     */
+    public function __construct(int|bool $availability, protected string $status = '')
+    {
+        $this->availability = (int)$availability;
+    }
+
+    /**
+     * Check if available.
+     *
+     * @return bool
+     */
+    public function isAvailable(): bool
+    {
+        return (bool)$this->availability;
+    }
+
+    /**
+     * Check if item has given availability status.
+     *
+     * @param int $availability Availability status
+     *
+     * @return bool
+     */
+    public function is(int $availability): bool
+    {
+        return $this->availability === $availability;
+    }
+
+    /**
+     * Check if status should be visible.
+     *
+     * @return bool
+     */
+    public function isVisible(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Set use unknown message
+     *
+     * @param bool $useUnknownMessage If unknown message shall be used
+     *
+     * @return void
+     */
+    public function setUseUnknownMessage(bool $useUnknownMessage): void
+    {
+        $this->useUnknownMessage = $useUnknownMessage;
+    }
+
+    /**
+     * Check if unknown message should be used.
+     *
+     * @return bool
+     */
+    public function useUnknownMessage(): bool
+    {
+        return $this->useUnknownMessage;
+    }
+
+    /**
+     * Get status description.
+     *
+     * @return string
+     */
+    public function getStatusDescription(): string
+    {
+        if (!empty($this->status)) {
+            return $this->status;
+        }
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'Unavailable';
+            case self::STATUS_AVAILABLE:
+                return 'Available';
+            default:
+                return 'Uncertain';
+        }
+    }
+
+    /**
+     * Get status label.
+     *
+     * @return string
+     */
+    public function getStatusLabel(): string
+    {
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'Checked Out';
+            case self::STATUS_AVAILABLE:
+                return 'Available';
+            default:
+                return 'HoldingStatus::availability_uncertain';
+        }
+    }
+
+    /**
+     * Get status class.
+     *
+     * @return string
+     */
+    public function getStatusClass(): string
+    {
+        switch ($this->availability) {
+            case self::STATUS_UNAVAILABLE:
+                return 'danger';
```

Comment:
I try to avoid coding CSS classes into server-side code, since it's theoretically possible that a single VuFind instance could have multiple themes that use different class names (e.g. if different libraries are involved). I realize this isn't very likely in practice, but it still might be a good idea to decouple the styling from the rest of the logic. Is it possible we need a view helper to encapsulate this logic rather than putting it here?